ir/translation/constraints.rs

```diff
@@ -370,6 +372,37 @@ fn add_typeql_isa(
     Ok(())
 }
 
+fn parse_iid(mut iid: &str) -> ByteArray<THING_VERTEX_MAX_LENGTH> {
+    fn from_hex(c: u8) -> u8 {
+        // relying on the fact that typeql ensures only hex digits
+        match c {
+            b'0'..=b'9' => c - b'0',
+            b'a'..=b'f' => c - b'a' + 10,
+            b'A'..=b'F' => c - b'A' + 10,
+            _ => unreachable!(),
+        }
+    }
+
+    iid = &iid["0x".len()..];
+
+    let mut bytes = [0u8; THING_VERTEX_MAX_LENGTH];
+    for (i, (hi, lo)) in iid.bytes().tuples().enumerate() {
+        bytes[i] = (from_hex(hi) << 4) + from_hex(lo);
+    }
+    let len = iid.as_bytes().len() / 2;
+    ByteArray::Inline(ByteArrayInline::new(bytes, len))
+}
```

Comment:
I suppose it doesn't hurt us to accept malformed (eg, too long or too short IIDs) since they will just fail comparison in the traversal. However, this can fail if the IID is 1 character long right? Because `iid["0x".len()..]` range will fail?

Reply:
No, but we will get an empty IID
Here we'll get `iid = "0"; len = 1 / 2 = 0;`
It will be rejected during execution (length checks will fail first, so no vertex will be constructed).